src/aap_eda/analytics/analytics_collectors.py

```diff
@@ -866,12 +747,66 @@ def _get_audit_action_qs(since: datetime, until: datetime):
     return audit_actions
 
 
-def _copy_table(table, query, path):
-    file_path = os.path.join(path, table + "_table.csv")
-    file = CsvFileSplitter(filespec=file_path)
-    with connection.cursor() as cursor:
-        with cursor.copy(query) as copy:
-            while data := copy.read():
-                byte_data = bytes(data)
-                file.write(byte_data.decode())
-    return file.file_list()
+def _copy_table(
+    table: str, queryset: QuerySet, path: str, buffer_size: int = 1024 * 1024
+) -> Optional[List[str]]:
+    """Export large datasets to CSV files using PostgreSQL COPY.
+
+    Args:
+        table (str): Source table name for file naming
+        query (QuerySet): Django QuerySet generating the COPY command, include:
+                          - Filter conditions
+                          - Selected fields
+                          Example: Model.objects.filter(...).values(...)
+        path (str): Output directory path with write permissions
+        buffer_size (int): In-memory buffer size in bytes.
+                           (Default: 1048576 = 1MB)
+
+    Returns:
+        Optional[List[str]]:
+        On success: List of chunked file paths
+        On failure: None
+
+    Raises:
+        DatabaseError: Connection/query failures
+        IOError: Filesystem permission issues
+    """
+    try:
+        view_name = f"temp_{table}_{uuid.uuid4().hex[:8]}"
+        file_path = os.path.join(path, table + "_table.csv")
+        file = CsvFileSplitter(filespec=file_path)
+
+        with connection.cursor() as cursor:
+            sql, params = queryset.query.sql_with_params()
+            create_view_sql = f"""
+                CREATE TEMPORARY VIEW {view_name} AS
+                {sql}
+            """
+            cursor.execute(create_view_sql, params)
+
+            copy_sql = f"""
+                COPY (SELECT * FROM {view_name})
+                TO STDOUT WITH CSV HEADER
+            """
+            with cursor.copy(copy_sql) as copy:
+                while True:
+                    data = copy.read()
+                    if not data:
+                        break
+                    byte_data = bytes(data)
+                    # Process data in chunks of buffer_size
+                    for i in range(0, len(byte_data), buffer_size):
+                        chunk = byte_data[i : i + buffer_size]
+                        file.write(chunk.decode())
+
+            cursor.execute(f"DROP VIEW IF EXISTS {view_name}")
+        return file.file_list()
+    except DatabaseError as e:
```

Comment:
Are you sure we want to pass on these errors?

Reply:
@Alex-Izquierdo This function is repeatedly used during file dumping by periodic or manual tasks. If an error occurs, IMO we should log it and let the task continue.